src/Raven.Server/Documents/Indexes/MapReduce/Static/MapReduceIndex.cs

```diff
@@ -53,14 +53,17 @@ protected MapReduceIndex(MapReduceIndexDefinition definition, StaticIndexBase co
 
         public override bool IsMultiMap => _compiled.Maps.Count > 1 || _compiled.Maps.Any(x => x.Value.Count > 1);
 
-        public override void ResetIsSideBySideAfterReplacement()
+        protected override void DisposeIndex()
         {
-            _isSideBySide = null;
+            base.DisposeIndex();
+
+            if (Type.IsJavaScript())
+                DocumentDatabase?.IndexStore?.Cache.Remove(_compiled.CacheKey);
```

Comment:
If you now remove it from the cache on shutdown, what use is the cache in the first place?